OpenRA.Mods.Common/Traits/Buildings/BuildingInfluence.cs

```diff
@@ -23,30 +23,36 @@ public class BuildingInfluenceInfo : ITraitInfo
 	public class BuildingInfluence
 	{
 		readonly Map map;
-		readonly CellLayer<Actor> influence;
+		readonly CellLayer<List<Actor>> influence;
 
 		public BuildingInfluence(World world)
 		{
 			map = world.Map;
 
-			influence = new CellLayer<Actor>(map);
+			influence = new CellLayer<List<Actor>>(map);
 		}
 
-		internal void AddInfluence(Actor a, IEnumerable<CPos> tiles)
+		public void AddInfluence(Actor a, IEnumerable<CPos> tiles)
 		{
 			foreach (var u in tiles)
-				if (influence.Contains(u) && influence[u] == null)
-					influence[u] = a;
+			{
+				if (influence.Contains(u))
+				{
+					if (influence[u] == null)
+						influence[u] = new List<Actor>();
+					influence[u].Add(a);
+				}
+			}
 		}
 
-		internal void RemoveInfluence(Actor a, IEnumerable<CPos> tiles)
+		public void RemoveInfluence(Actor a, IEnumerable<CPos> tiles)
 		{
 			foreach (var u in tiles)
-				if (influence.Contains(u) && influence[u] == a)
-					influence[u] = null;
+				if (influence.Contains(u) && influence[u].Contains(a))
+					influence[u].Remove(a);
 		}
 
-		public Actor GetBuildingAt(CPos cell)
+		public List<Actor> GetBuildingsAt(CPos cell)
 		{
 			return influence.Contains(cell) ? influence[cell] : null;
```

Comment:
If you return IEnumerable<Actor> instead of List and return an empty array / yield break you can get rid of all the null checks before doing .Any on the return value of GetBuildingsAt